aptos-move/aptos-vm/src/verifier/resource_groups.rs

```diff
@@ -86,25 +91,51 @@ pub(crate) fn validate_module_and_extract_new_entries(
             (BTreeMap::new(), BTreeMap::new())
         };
 
-    let (original_groups, original_members) =
+    let (original_groups, original_members, mut structs) =
         extract_resource_group_metadata_from_module(session, &module.self_id())?;
 
     for (member, value) in original_members {
         // We don't need to re-validate new_members above.
         if Some(&value) != new_members.remove(&member).as_ref() {
-            metadata_validation_err("Invalid change in resource_group_member")?;
+            metadata_validation_err("Invalid removal of resource_group_member attribute")?;
```

Comment:
👍 (I'm still confused under what conditions the message goes to the txn info, but as long as it replays)

Reply:
I'm in the same boat re:messages

Reply:
messages are there only for API and logs at this point in time... and I'm not entirely sure if we surface it to the API... it doesn't get put in persistent storage, I think it did at one point and you @msmouse fixed that.

Reply:
no I didn't

Reply:
well someone did....